pennylane/tape/qnode.py

```diff
@@ -513,13 +519,23 @@ def construct(self, args, kwargs):
             )
 
     def __call__(self, *args, **kwargs):
+
+        # If shots specified from call, we override the device's default shots value.
+        shots = kwargs.pop("shots", None)
+        if shots is not None:
+            original_shots = self.device.shots  # remember device shots
+            self.device.shots = shots  # temporarily change shots

```

Comment:
This is fine for now, but we _may_ have to revisit later when `self.device.shots` is a list/tuple. Since this won't copy the original list, but will just be a reference!

Reply:
Yes, implementing that list part is still a major challenge...